Improve grammar, spelling, and punctuation in Usage guide section (#9395)
wagtail/wagtail: docs/topics/search/indexing.md

### Example

> This creates an `EventPage` model with two fields: `description` and `date`. The `description` is indexed as a `SearchField` and `date` is indexed as a `FilterField`.

Reviewer: `description` is the name of the field, so it's correct without "The".
```suggestion
This creates an `EventPage` model with two fields: `description` and `date`. `description` is indexed as a `SearchField` and `date` is indexed as a `FilterField`.
```

-   **boost** (`int/float`) - This allows you to set fields as being more important than others. Setting this to a high number on a field will cause pages with matches in that field to be ranked higher. By default, this is set to 2 on the Page title field and 1 on all other fields.

    ```{note}
    The PostgresSQL full-text search only supports [four weight levels (A, B, C, D)](https://www.postgresql.org/docs/current/textsearch-features.html).
    When the database search backend `wagtail.search.backends.database` is used on a PostgreSQL database, it will take all boost values in the project into consideration and group them into the four available weights.

    This means that in this configuration there are effectively only four boost levels used for ranking the search results, even if more boost values have been used.

>     You can find out roughly which boost thresholds map to which weight in PostgresSQL by starting a new Django shell with `./manage.py shell` and inspecting `wagtail.search.backends.database.postgres.weights.BOOST_WEIGHTS`.

Reviewer: Fixing an extra typo while we're at it... PostgresSQL should be PostgreSQL.
```suggestion
    You can find out roughly which boost thresholds map to which weight in PostgreSQL by starting a new Django shell with `./manage.py shell` and inspecting `wagtail.search.backends.database.postgres.weights.BOOST_WEIGHTS`.
```

### `index.AutocompleteField`

These are used for autocomplete queries that match partial words. For example, a page titled `Hello World!` will be found if the user only types `Hel` into the search box.

This takes the exact same options as `index.SearchField` (with the exception of `partial_match`, which has no effect).

```{note}

> Only index fields are displayed in the search results with ``index.AutocompleteField``. This allows users to see any words that were partial-matched.

Reviewer: I think "index" is meant to be the verb here - as in, "you should only index these fields".

```suggestion
``index.AutocompleteField`` should only be used on fields that are displayed in the search results. This allows users to see any words that were partial-matched.
```
